### PR TITLE
Upgrade to PHP 8.2

### PR DESCRIPTION
**Pull request by eldadfux**
## What does this PR do?

Upgrades Appwrite to run on PHP 8.2

## Test Plan

Existing tests. 

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: tests/e2e/Services/GraphQL/AvatarsTest.php [hunk 1]
```diff
@@ -29,7 +29,7 @@ public function testGetCreditCardIcon()
             'x-appwrite-project' => $projectId,
         ], $this->getHeaders()), $graphQLPayload);
 
-        $this->assertEquals(18767, \strlen($creditCardIcon['body']));
+        $this->assertEquals(18546, \strlen($creditCardIcon['body']));
```
Comment (abnegate) [inline tests/e2e/Services/GraphQL/AvatarsTest.php:32]: Any idea why these all changed 🤔

File: tests/resources/initials.png
Comment (abnegate) [inline tests/resources/initials.png:1]: In what way did the images change? They look the same but all have different sizes

Comment (PineappleIOnic) [reply tests/resources/initials.png:1]: I ran it through an image compare tool https://www.img2go.com/result#j=66ec9efe-b732-49fa-8250-ffe104775df2 doesn't seem to have any differences. Maybe a compression thing?

**Review by abnegate — APPROVED** [review]

**Comment by fogelito** [comment]
Remove "league/csv": "9.7.1"
Getting this error onAppwrite
Fatal error: Type of League\Csv\CharsetConverter::$filtername must be string (as in class php_user_filter) in /usr/src/code/vendor/league/csv/src/CharsetConverter.php on line 37

**Comment by abnegate** [comment]
> Remove "league/csv": "9.7.1" Getting this error onAppwrite Fatal error: Type of League\Csv\CharsetConverter::$filtername must be string (as in class php_user_filter) in /usr/src/code/vendor/league/csv/src/CharsetConverter.php on line 37

@fogelito Looks like version 9.7 only supported up to PHP 8.0, a newer version supports 8.2, will update to that :+1:

**Pull request merged**